autovalet_navigation/scripts/autovalet_navigation/move_base_listener.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python
+
+import numpy as np
+
+# Include ROS libs
+import rospy
+import tf2_ros
+from tf2_geometry_msgs import do_transform_pose
+
+# Include messages
+from geometry_msgs.msg import PoseStamped, Quaternion, Pose
+from tf.transformations import quaternion_matrix, quaternion_from_matrix
+from actionlib_msgs.msg import GoalStatusArray
+
+def getCurrentPose():
+
+    tf_buffer      = tf2_ros.Buffer(rospy.Duration(1200.0)) # Length of tf2 buffer (?)
+    tf_listener    = tf2_ros.TransformListener(tf_buffer)
+
+
+    transform = tf_buffer.lookup_transform("icp_odom", # target_frame_id
+                                "map", #source frame
+                                rospy.Time(0), #get the tf at first available time
+                                rospy.Duration(1.0)) #timeout after 1
+
+    current_pose = PoseStamped()
+    current_pose.header = transform.header
+    current_pose.pose.position.x = transform.transform.translation.x
+    current_pose.pose.position.y = transform.transform.translation.y
+    current_pose.pose.position.z = transform.transform.translation.z
+    current_pose.pose.orientation = transform.transform.rotation
+
+    return current_pose
+
+def arePosesClose(pose1, pose2):
+    result = True
+
+    if result and pose1.pose.position.x != pose2.pose.position.x:
+        result = False
+    if result and pose1.pose.position.y != pose2.pose.position.y:
+        result = False
+
+    return result
+
+class State:
+    Waiting = 0
+    Planning = 1
+    Executing = 2
+    Fail = 3
+    Success = 4
+
+class MoveBaseListener:
+    def __init__(self, debug=True):
+        self.goal_sub       = rospy.Subscriber("/move_base_simple/goal", PoseStamped, self.goalCallback)
+        self.status_sub     = rospy.Subscriber("/move_base/status", GoalStatusArray, self.MBstatusCallback)
+        
+        self.goal_pose      = PoseStamped()
+        self.start_pose     = PoseStamped()
+
+        self.plan_state     = State.Waiting
+
+        self.debug          = debug
+
+    def getState(self):
+        output = None
+        if self.plan_state == State.Waiting:
+            output = "waiting"
+        elif self.plan_state == State.Planning:
+            output = "planning"
+        elif self.plan_state == State.Executing:
+            output = "executing"
+        elif self.plan_state == State.Fail:
+            output = "fail"
+        elif self.plan_state == State.Success:
+            output = "success"
```

Comment:
I think it's a bit redundant here.... would renaming the class `State` as `MoveBaseState` and we will only output `self.plan_state` better? modules outside can compare the state with `MoveBaseState.blahblahblah`.

Reply:
yea u right. i initially did the string thing cause i knew i had `State` in both places. but that was super hacky. this is better. fixed